boards/teensy4/README.md

```diff
@@ -0,0 +1,74 @@
+Teensy 4 Development Board
+==========================
+
+The `teensy4` board supports both the Teensy 4.0 and Teensy 4.1 form factors.
+For more information, visit the product pages:
+
+- [Teensy 4.0 Development Board](https://www.pjrc.com/store/teensy40.html)
+- [Teensy 4.1 Development Board](https://www.pjrc.com/store/teensy41.html)
```

Comment:
It looks like (based on https://www.pjrc.com/store/teensy41.html) these two boards vary quite a bit. Can we choose just one that this has been tested on? Otherwise it's not clear how the linker file should be set, or if it is ok to add an sdcard driver.

Reply:
I updated the README to state that we guarantee support for the Teensy **4.0** board. The `teensy4` crate will never include features that are only for the 4.1 board. I've left more thoughts in f981a847's commit message.

Would you prefer to rename the crate to `teensy40` as a stronger support signal? Or, could there be a future where the `teensy4` board supports the common functions between 4.0 and 4.1, and we use cargo feature flags to enable 4.1 features?

Reply:
Typically, Tock avoids cargo feature flags for enabling features specific to a particular chip within a family of supported chips. Instead, Tock structures modules in a specific way so that more specific chips inherit from a library of shared functionality, re-exporting from the shared library all functionality that is unchanged by the more specific chip. For the best example of this, see `chips/nrf5x', `chips/nrf52`, and `chips/nrf52840`.

Given that you currently don't have any features only supported by one chip and not another, and have only tested on a teensy 4.0, I think it is best to name the chip crate `teensy40`. We can revisit that if explicit, tested support for the teensy 4.1 is added later.